astropy/coordinates/sky_coordinate.py

```diff
@@ -484,6 +487,94 @@ def transform_to(self, frame, merge_attributes=True):
             frame_kwargs.pop(attr)
         return self.__class__(new_coord, **frame_kwargs)
 
+    def apply_space_motion(self, new_obstime=None, dt=None):
+        """
+        Update the position of the source represented by this coordinate object
```

Comment:
Needs a `Returns` section.  Along similar lines, I'd say change the first word from "Update" to "Returns a new SkyCoord for" or something like that.  "update" to me sounds like "inplace".